src/librbd/Operations.cc

```diff
@@ -779,7 +779,9 @@ int Operations<I>::snap_rollback(const cls::rbd::SnapshotNamespace& snap_namespa
   }
 
   r = prepare_image_update();
-  if (r < 0) {
+  if (r == -EAGAIN) {
```

Comment:
@dillaman, this case is that no real client use image but the exclusive lock is in-use when the user first try, the checking interval is 30s. Sorry, I don't know why this -EAGAIN returned by exclusive lock code is changed to -EROFS finally.